yrs/src/types/mod.rs

```diff
@@ -220,14 +229,40 @@ pub struct Branch {
     /// A tag name identifier, used only by [XmlElement].
     pub name: Option<String>,
 
-    pub item: Option<BlockPtr>, //TODO: isn't this equivalent to `ptr` field?
-
     /// A length of an indexed sequence component of a current branch node. Map component elements
     /// are computed on demand.
     pub len: u32,
 
     /// An identifier of an underlying complex data type (eg. is it an Array or a Map).
     type_ref: TypeRefs,
+
+    observers: Option<EventHandler<Event>>,
+}
+
+impl std::fmt::Debug for Branch {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Branch")
+            .field("start", &self.start)
+            .field("map", &self.map)
+            .field("ptr", &self.ptr)
+            .field("name", &self.name)
+            .field("len", &self.len)
+            .field("type_ref", &self.type_ref)
+            .finish()
+    }
+}
+
+impl Eq for Branch {}
+
+impl PartialEq for Branch {
```

Comment:
Since `Branch` contains event handlers now, derivation of structural equality is no longer possible.